### DISCUSSION TITLE
Check expiration callback

### DISCUSSION THREAD

**Discussion by ddumitre7** [jellydator/ttlcache#85] [Polls]
I would a need to set a callback that will be called when an item is about to expire in order to allow external code to decide whether the item expires or remains for another TTL cycle.

My use-case is straightforward: At TTL expiration, if the value still exists in the database, then leave the corresponding item in the cache. Otherwise evict it

Prior art:

https://pkg.go.dev/github.com/ReneKroon/ttlcache/v2#Cache.SetCheckExpirationCallback

**Comment by swithek** [comment]
Having a callback like that is a good idea, however, I'm not yet sure if it should be allowed to block the cache's execution since this would affect other operations like `Set()` or `Get()` (when iterating over the expired items, the global cache [mutex is locked](https://github.com/jellydator/ttlcache/blob/40f09cf4bc54b9a74ead0a3843e650a3d387e689/cache.go#L306)). If we were to add this feature, the user would have to be aware that their code/callback might degrade the performance of the cache itself. Perhaps with proper documentation this could work?
